agent/api/task/task_linux.go

```diff
@@ -355,3 +357,38 @@ func (task *Task) BuildCNIConfig(includeIPAMConfig bool, cniConfig *ecscni.Confi
 
 	return cniConfig, nil
 }
+
+// BuildCNIConfigBridgeMode builds a list of CNI network configurations for a task in docker bridge mode.
+func (task *Task) BuildCNIConfigBridgeMode(cniConfig *ecscni.Config, containerName string) (*ecscni.Config, error) {
```

Comment:
nit: I see all the methods are marked public. why?

Reply:
Making `BuildCNIConfigAwsvpc` and `BuildCNIConfigBridgeMode` public and than they can be used in the engine package, like:

`agent/engine/docker_task_engine.go`
* buildCNIConfigFromTaskContainerAwsvpc() [[L1940](https://github.com/yinyic/amazon-ecs-agent/blob/6a66b11cbc4f727012a78b35470cad773e836454/agent/engine/docker_task_engine.go#L1940)]

* buildCNIConfigFromTaskContainerBridgeMode() [[L1958](https://github.com/yinyic/amazon-ecs-agent/blob/6a66b11cbc4f727012a78b35470cad773e836454/agent/engine/docker_task_engine.go#L1958)]

`agent/engine/task_manager.go` 
* releaseIPInIPAM()[[L757](https://github.com/yinyic/amazon-ecs-agent/blob/6a66b11cbc4f727012a78b35470cad773e836454/agent/engine/task_manager.go#L757)]

Reply:
Thanks ChienHan! 
Like ChienHan mentioned, docker engine drives the container lifecycle as a state machine, and task package owns most of the actions to take during each task/container state.